eda/src/autogluon/eda/analysis/model.py

```diff
@@ -0,0 +1,90 @@
+from typing import Union, List, Optional
+
+from sklearn.metrics import confusion_matrix
+
+from autogluon.core.constants import BINARY, MULTICLASS
+from autogluon.tabular import TabularPredictor
+from .base import AnalysisState, AbstractAnalysis
+
+__all__ = ["AutoGluonModelEvaluator"]
+
+
+class AutoGluonModelEvaluator(AbstractAnalysis):
+    """
+    Evaluate AutoGluon model performance.
+
+    This analysis requires trained classifier passed in `model` arg and uses 'val_data' dataset to asses model performance.
+
+    Parameters
+    ----------
+    model: TabularPredictor, required
+        fitted AutoGluon model to analyze
+    val_data: pd.DataFrame, required
```

Comment:
[minor] Quick verification. I assume the validation dataset should follow the same column names seen by the model. Most likely it would be the case, better to call out in the docstring in case users generate the validation datasets using a different process

Reply:
There are two things we can do here:
1. Sanity check.
2. Add another component to do this split for the users with a wrapper - similar to how `Sampler` is used.